lib/charms/tls_certificates_interface/v4/tls_certificates.py

```diff
@@ -1014,6 +1016,17 @@ def __init__(
             certificate_requests (List[CertificateRequestAttributes]):
                 A list with the attributes of the certificate requests.
             mode (Mode): Whether to use unit or app certificates mode. Default is Mode.UNIT.
+                In UNIT mode the requirer will place the csr in the unit relation data.
+                Each unit will have its own private key,
+                certificate signing request and certificate.
+                A common use case for UNIT mode is internal communication between units.
+                In APP mode the requirer will place the csr in the app relation data.
+                The leader unit will be responsible for
+                placing the csr in the app relation databag.
+                A common use case for APP mode
+                is when the underlying application takes care of the certificate
+                For example distributing it or using it for admin purposes.
```

Comment:
This use case is not clear to me, can you please be more specific with potentially a real example.

Reply:
Tried to make it a bit more explicit